### PR TITLE
Improve compilation of the target-table filters

### PR DESCRIPTION
**Pull request by smondet**
Now `(not ...)` expressions are explored further but it is still not
perfect because the protocol does not allow to express all the filters.

Changing the protocol to be complete is left for future work.

This fixes in particular the issue #288, the filter `(not (all))` gets
transformed to `` `Created_after max_float`` which is about 5.7 × 10^300
years in the future.

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/ketrew/294)

### CODE REVIEW

File: src/client-joo/target_table.ml [hunk 1]
```diff
@@ -274,8 +274,19 @@ module Filter = struct
             | Some old, Some `All -> Some `All
             | Some (`Created_after t1), Some (`Created_after t2) ->
               Some (`Created_after (min t1 t2)))
-      | `Not _ ->
-        None (* we reach the limits of this weird logic … *)
```
Comment (ihodes) [inline src/client-joo/target_table.ml]: lol

File: src/client-joo/target_table.ml [hunk 2]
```diff
@@ -274,8 +274,19 @@ module Filter = struct
...
+      | `Not something ->
+        begin match something with
+        | `Has_tags _
+        | `Name _ | `Id _
+        | `Status _ -> None
+        | `All -> Some (`Created_after max_float)
```
Comment (ihodes) [inline src/client-joo/target_table.ml]: Why wouldn't this just be `None` as well?

**Comment by ihodes** [comment]
Just one question, LGTM to me!

File: src/client-joo/target_table.ml (hunk 2 above)
Comment (smondet) [inline src/client-joo/target_table.ml]: That was the previous behavior which made `(not (all))` be `None` so the UI would show the same result as `(created-after 1970)` (which is in practice is `(all)`).

Now `(not (all))` shows nothing (by hackishly asking for anything created after the end of the universe).

**Pull request merged**